pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -93,6 +91,7 @@ func (c CertRotationController) sync() error {
 		newCondition.Message = syncErr.Error()
 	}
 	if _, _, updateErr := v1helpers.UpdateStaticPodStatus(c.OperatorClient, v1helpers.UpdateStaticPodConditionFn(newCondition)); updateErr != nil {
+		syncCtx.Recorder().Warningf("RotationError", newCondition.Message)
```

Comment:
does this really need an event?

Reply:
wrong code... I should get `updated` bool and only fire this event once when we update the condition.

Reply:
fixed (locally):

```golang
	_, updated, updateErr := v1helpers.UpdateStaticPodStatus(c.OperatorClient, v1helpers.UpdateStaticPodConditionFn(newCondition))
	if updateErr != nil {
		return updateErr
	}
	if updated && syncErr != nil {
		syncCtx.Recorder().Warningf("RotationError", newCondition.Message)
	}
```

Reply:
am not convinced we want to create events on every error.

Reply:
events make timelines

Reply:
logs are timelines.